numpy/lib/tests/test_utils.py

```diff
@@ -176,3 +186,46 @@ def _has_method_heading(cls):
 
     assert _has_method_heading(WithPublicMethods)
     assert not _has_method_heading(NoPublicMethods)
+
+
+def _compare_dtypes(dt1, dt2):
+    ret = True
+    # parse attrs except descr which can contains metadata infos
+    for k in dir(dt1):
+        if not ((k.startswith('__') and k.endswith('__'))
+                or k in ['descr', 'metadata', 'newbyteorder']):
+            ret &= (getattr(dt1, k) == getattr(dt2, k))
```

Comment:
I would write this as:
```
for ...:
    if getattr(dt1, k) != getattr(dt2, k):
        return False

return True
```

Reply:
You could also check `np.can_can(dt1, dt2, casting="no")`, since that should only really work only metadata mismatches.  (The only issue should be e.g. `"long"` and `"longlong"` which on most systems, except windows, are the same).

So, I am fine with this also, it is at least thorough and easy enough to read.